File: webservername-server/src/main/java/com/mycompany/webservername/server/initializer/AppInitializer.java
package com.mycompany.webservername.server.initializer;

import com.mycompany.webservername.server.config.AppContextConfig;
import com.mycompany.webservername.server.config.DispatcherConfig;
import org.springframework.web.WebApplicationInitializer;
import org.springframework.web.context.ContextLoaderListener;
import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
import org.springframework.web.servlet.DispatcherServlet;

import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.ServletRegistration;

/**
 * Created by Artemie on 26.08.2016.
 */
public class AppInitializer implements WebApplicationInitializer {
    @Override
    public void onStartup(ServletContext servletContext) throws ServletException {
        AnnotationConfigWebApplicationContext rootContext =
                new AnnotationConfigWebApplicationContext();
        rootContext.register(AppContextConfig.class);

        servletContext.addListener(new ContextLoaderListener(rootContext));

        AnnotationConfigWebApplicationContext dispatcherContext =
                new AnnotationConfigWebApplicationContext();
        dispatcherContext.register(DispatcherConfig.class);

        ServletRegistration.Dynamic dispatcher =
                servletContext.addServlet("dispatcher", new DispatcherServlet(dispatcherContext));
        dispatcher.setLoadOnStartup(1);
        dispatcher.addMapping("/*");
    }

}

File: webservername-server/src/main/kotlin/com/mycompany/servername/server/kotlinbased/initializer/AppInitializer.kt
package com.mycompany.servername.server.kotlinbased.initializer

import com.mycompany.servername.server.kotlinbased.config.AppContextConfig
import org.springframework.web.WebApplicationInitializer
import org.springframework.web.context.ContextLoaderListener
import org.springframework.web.context.support.AnnotationConfigWebApplicationContext
import org.springframework.web.servlet.DispatcherServlet
import javax.servlet.ServletContext
import javax.servlet.ServletException

/**
 * Created by Artemie on 26.08.2016.
 */
class AppInitializer : WebApplicationInitializer {
    @Throws(ServletException::class)
    override fun onStartup(servletContext: ServletContext) {
        val rootContext = AnnotationConfigWebApplicationContext()
        rootContext.register(AppContextConfig::class.java)

        servletContext.addListener(ContextLoaderListener(rootContext))

        val dispatcherContext = AnnotationConfigWebApplicationContext()
        dispatcherContext.register(DispatcherConfig::class.java)

        val dispatcher = servletContext.addServlet("dispatcher", DispatcherServlet(dispatcherContext))
        dispatcher.setLoadOnStartup(1)
        dispatcher.addMapping("/*")
    }

}
